Sources/Tokenizers/PreTokenizer.swift

```diff
@@ -277,6 +276,42 @@ extension String {
         return result
     }
 
+    /// This version supports capture groups, wheres the one above doesn't
+    func split(by captureRegex: NSRegularExpression) -> [String] {
```

Comment:
This was a bit annoying, it's just another version of the previous `split` helper we wrote that works with capture groups.